IPs/RSKIP_tbd.md

```diff
@@ -0,0 +1,229 @@
+        - Create it as described [here](#proof-transaction)
+        - Add it to a new storage set, key `spendTxWaitingForSignatures`
+        - Set it in a new Storage entry:
+            ```
+            KEY: "SPEND_TX_HASH"
+            VALUE: hash of SVP proof tx
+            ```    
+
+##### Proof transaction
+
+The Proof transaction is a Bitcoin transaction where a UTXO from the proposed Powpeg is sent to the current Powpeg.
+This transaction will be as simple and small as possible to avoid spending much satoshis as fees.
+
+Proof transaction structure:
+```
+{
+    inputs: [
+        {
+            txid: FUND_TX_HASH_SIGNED,
+            output_index: FUNDING_TX_OUTPUT_INDEX
+        }
+    ],
+    outputs: [
+        {
+            script: CURRENT_POWPEG_OUTPUT_SCRIPT,
+            value: TOTAL_UTXO - FEES
+        }
+    ]
+}
+```
+
+#### Proof transaction confirmation
+
+The SVP proof transaction should go through a new "peg-out signing process", since it has to be signed for the proposed pegnatories which belong neither the active nor the retiring powpeg, but the proposed one.
+
+#### Proof transaction signing
+
+The proof transaction should be signed as any other peg-out, but by the proposed powpeg. For this to happen, a new Bridge method `ADD_SPEND_TX_SIGNATURE` should be created to perform the exact same actions as the `ADD_SIGNATURE` but to accept signatures from public keys that don't belong to an active/retiring Powpeg (let's recall the proposed Powpeg is not active at this point).
+The `addSpendTxSignature` method should check that the public key belongs to the proposed Powpeg and that the hash of the transaction intended to be signed exists in the `SPEND_TX_HASH` storage entry.
+
+Once fully signed, the Bridge should remove the entry from the `spendTxWaitingForSignatures` and emit the release_btc event as usual.
+
+### Proof verification
+
+With the SVP proof transaction broadcasted, the Bridge now needs to validate it and proceed with the election commitment as expected.
+
+The proof transaction should be registered through the registerBtcTransaction method. As before, it should verify:
+- SVP period is ongoing
+- Transaction hash without signatures is registered in the `SPEND_TX_HASH` storage key.
+
+If the hash matches the one registered in the storage, then the SVP process is considered completed and we can proceed with the committment of the proposed -and now validated- Powpeg.
+
+### Validation phase duration
+
+The duration of this phase must account for:
+- Spending a UTXO from the current Powpeg (Bridge confirmations + pegnatories signing)
+- Confirming the UTXO on Bitcoin (Bitcoin confirmations for the UTXO to be accepted)
+- Sending back the UTXO to the current Powpeg (Proposed pegnatories signing)
+
+The recommendation is that this phase takes at least 3 times the blocks a peg-out confirmation has. **TODO: CONFIRM THIS**
```

Comment:
What's the rational behind this? Wouldn't it take long anyways?

1) The active fed will send some funds to the proposed fed. That will take 4,000 rsk blocks + 100 bitcoin blocks to get the utxos registered in the Bridge under the proposed federation.
2) The proposed federation will send the funds back to the active federation as the proof it can spend them. 4,000 + 100 bitcoin blocks more.

For a total of 8k+ rsk blocks and 200+ bitcoin blocks.

Since the rsk and bitcoin blocks confirmations happen sequentially, around 16,000 rsk block will be mined from start to finish during this process without doing anything special. Since when the first 4,000 rsk blocks are mined, then 100 bitcoin blocks will be mined at the same time that the rsk blockchain will keep mining around 4,000 rsk blocks more for when the first utxo to the proposed fed gets to the Bridge, so around 8k rsk blocks for the first step.
Then around 8k more rsk blocks for the proposed fed to send the funds back to the active fed.

So, with all this in mine, do we want to manually add more waiting blocks?

Reply:
Now I understand.

I used the block estimator tools to compare. 

I used the rsk estimator tool: https://fluffy-system-o4omlo9.pages.github.io/#rskEstimatorContainer

And the btc estimator tool: https://fluffy-system-o4omlo9.pages.github.io/#btcEstimatorContainer

As of now, in rsk and bitcoin mainnet, it takes about 1 day and 8 hours for 4000 rsk blocks to be mined, and about 14 hours for 100 bitcoin blocks to be mined.

So, when the active federation creates a validation pegout (similar to a migration pegout) to the proposed federation, this pegout will wait for a minimum of 4000 rsk blocks. Then, since this acts like a migration pegout or a change registration, since it is directed to a "federation", that pegout will wait 100 bitcoin blocks, which is about 1681 rsk blocks. So, so far, from the rsk perspective, from the creation of this validation pegout to the Bridge registering these utxos to the proposed federation, a minimum of 4000 + 1681 = 5681 rsk blocks will have passed.

Then, the process is repeated again, but from the perspective of the proposed federation. The proposed federation will create a pegout to the active federation to demonstrate that it can actually spend funds. This pegout will be treated as usual: it will wait 4000 rsk blocks for confirmations, then 100 bitcoin blocks for it to be registered back to the Bridge under the active deration. So, another 4000 + 1681 = 5681 rsk blocks.

So, it would take a minimum of 5681 + 5681 = 11362 rsk blocks for the whole process to be done.

From this number, we can then add some more blocks `federationValidationPeriod` just in case this process actually takes more blocks.

So, what will be the actual value of `federationValidationPeriod` now that we know the minimum number of rsk blocks the whole process will take? 12000 might not be enough in this case. Should we add another 4000 rsk blocks to the 11000+ blocks in my calculation? Let's discuss it.